insights/core/__init__.py

```diff
@@ -585,15 +595,50 @@ def replacer(match):
                 return format_conversion_for[match.group(1)]
             else:
                 raise ParseException(
-                    "_get_after does not understand strptime format '{c}'".format(
+                    "get_after does not understand strptime format '{c}'".format(
                         c=match.group(0)
                     )
                 )
 
         # Please do not attempt to be tricky and put a regular expression
         # inside your time format, as we are going to also use it in
         # strptime too and that may not work out so well.
-        time_re = re.compile('(' + timefmt_re.sub(replacer, time_format) + ')')
+
+        # Check time_format - must be string or list.  Set the 'logs_have_year'
+        # flag and timestamp parser function appropriately.
+        # Grab values of dict as a list first
+        if type(time_format) == dict:
```

Comment:
Fair enough, although I really wonder why anyone would be subclassing a dict for this argument :-)  `isinstance()` isn't perfect either, but it's better than `type()`.